frontend/passkeys.js

```diff
@@ -3,118 +3,209 @@ import { getPasskeyNameFromUser } from './utils/modalUtils.js';
-         * A null value means any authenticator is allowed.
-         * Unfortunately, even though null is the recommended value, Safari won't allow it
-         * and Firefox gives a warning about it.
-         * So we remove the field if its value is null and all browsers are then happy.
-         * The effect is unchanged.
-         *
-         * This is a temporary measure until browsers accept the null value.
-         */
-        if (regOptionsResponseJson.authenticatorSelection?.authenticatorAttachment === null) {
-            delete regOptionsResponseJson.authenticatorSelection.authenticatorAttachment;
-        }
+      if (!userConfirmed) {
+        displayToast('Passkey registration cancelled', messageType.info);
+        return;
+      }
 
-        // 4. Create a new passkey
-        const credentialCreationOptions = PublicKeyCredential.parseCreationOptionsFromJSON(regOptionsResponseJson);
-        const createdCredential = await navigator.credentials.create({ publicKey: credentialCreationOptions });
-        let passkeyName;
-        try {
-            passkeyName = await getPasskeyNameFromUser();
-            if (passkeyName === null) {
-                return;
-            }
-        } catch (error) {
-            console.error('Modal error:', error);
-            throw error;
-        }
+      // Now proceed with the authentication flow
+      existingCredential = await passkeyApi.getUserCredential(authOptionsJson);
 
-        // 5. Make the registration call - includes authentication credentials if they exist so that they can be verified
-        if (existingCredential) {
-            createAndSubmitForm('/passkey/register', {
-                credentials: JSON.stringify(existingCredential.toJSON()),
-                passkey: JSON.stringify(createdCredential.toJSON()),
-                csrfToken: csrfToken,
-                passkeyName: passkeyName
-            });
-        } else {
-            createAndSubmitForm('/passkey/register', {
-                passkey: JSON.stringify(createdCredential.toJSON()),
-                csrfToken: csrfToken,
-                passkeyName: passkeyName
-            });
-        }
-    } catch (err) {
-        if (err.name === 'InvalidStateError') {
-            console.warn('Passkey already registered:', err);
-            displayToast('This passkey has already been registered.', messageType.warning);
-        } else {
-            console.error('Error during passkey registration:', err);
-            displayToast('Passkey not registered', messageType.error);
-        }
+      await showConfirmationModal(
+        'Register new passkey',
+        'Authentication successful! Now register your new passkey.',
+        'register'
+      );
+    }
+
+    // 3. Fetch the passkey creation options
+    const regOptionsResponse = await passkeyApi.fetchWithCsrf(
+      '/passkey/registration-options',
+      'POST',
+      null,
+      csrfToken
+    );
```

Comment:
Got the same error as above here too: 
```suggestion
    const regOptionsResponse = await passkeyApi.fetchWithCsrf(
      '/passkey/registration-options',
      'POST',
      null,
      null,
      csrfToken
    );
```